lib/attrsets.nix

```diff
@@ -913,6 +913,24 @@ rec {
     ) [{}] (attrNames attrsOfLists);
 
 
+  /**
+    Return the result of function f applied to the cartesian product of attribute set value combinations.
+    Equivalent to using cartesianProduct followed by map.
+
```

Comment:
Needs an `Inputs` and `Type` section, see the other functions in this file (though this is [very recent](https://github.com/NixOS/nixpkgs/pull/296186))